modules/apps/transfer/types/msgs.go

```diff
@@ -69,10 +69,22 @@ func NewMsgTransfer(
 // NOTE: The recipient addresses format is not validated as the format defined by
 // the chain is not known to IBC.
 func (msg MsgTransfer) ValidateBasic() error {
-	if err := validateSourcePortAndChannel(msg); err != nil {
-		return err // The actual error and its message are already wrapped in the called function.
+	if err := msg.validateForwarding(); err != nil {
+		return err
 	}
+	if !msg.Forwarding.Unwind {
```

Comment:
I wanted to achieve the same, but couldn't find a way that's actually nice.

The problem is that, if we move this inside `validateForwarding`, we are actually validating something that is not forwarding-related (since we have `len(hops)==0 && !msg.Forwarind.Unwind`) hidden there. 

I thought of the current situation as a valid compromise but I'm open to suggestion